pkg/auth/providers/oidc/oidc_client.go

```diff
@@ -6,35 +6,49 @@ import (
 	"crypto/x509"
 	"net/http"
 
+	"github.com/pkg/errors"
+
 	"github.com/coreos/go-oidc/v3/oidc"
 )
 
 func AddCertKeyToContext(ctx context.Context, certificate, key string) (context.Context, error) {
-	if certificate != "" && key != "" {
-		certKeyClient := &http.Client{}
-		err := GetClientWithCertKey(certKeyClient, certificate, key)
-		if err != nil {
-			return nil, err
-		}
-		return oidc.ClientContext(ctx, certKeyClient), nil
+	certKeyClient, err := GetClientWithCertKey(certificate, key)
+	if err != nil {
+		return nil, err
+	}
+	return oidc.ClientContext(ctx, certKeyClient), nil
+}
+
+func GetClientWithCertKey(certificate, key string) (*http.Client, error) {
+	// Use the default transport to pick up on things like http.ProxyFromEnvironment
+	transport := http.DefaultTransport.(*http.Transport).Clone()
+
+	tlsConfig, err := getTLSClientConfig(certificate, key)
+	if err != nil {
+		return &http.Client{Transport: transport}, err
 	}
-	return ctx, nil
+
+	transport.TLSClientConfig = tlsConfig
+	return &http.Client{Transport: transport}, nil
 }
 
-func GetClientWithCertKey(httpClient *http.Client, certificate, key string) error {
-	if certificate != "" && key != "" {
-		keyPair, err := tls.X509KeyPair([]byte(certificate), []byte(key))
-		if err != nil {
-			return err
-		}
-		caCertPool := x509.NewCertPool()
-		caCertPool.AppendCertsFromPEM([]byte(certificate))
-		httpClient.Transport = &http.Transport{
-			TLSClientConfig: &tls.Config{
-				RootCAs:      caCertPool,
-				Certificates: []tls.Certificate{keyPair},
-			},
-		}
+func getTLSClientConfig(certificate, key string) (*tls.Config, error) {
+	if certificate == "" || key == "" {
+		return nil, errors.New("certificate or key provided, but not both")
 	}
-	return nil
+
+	keyPair, err := tls.X509KeyPair([]byte(certificate), []byte(key))
+	if err != nil {
+		return nil, err
+	}
+
+	caCertPool, err := x509.SystemCertPool()
+	if err != nil {
+		return nil, err
+	}
+
+	return &tls.Config{
+		RootCAs:      caCertPool,
+		Certificates: []tls.Certificate{keyPair},
+	}, nil
```

Comment:
I also have a use case where we need to allow TLS Renegotiation for environments that require PKI through access cards and tokens. In these cases, the TLS protocol is TLS1.2 and renegotiation can't occur because by default, Go clients use TLS1.3. To fix this would be as simple as adding the following.
```suggestion
	return &tls.Config{
		RootCAs:      caCertPool,
		Certificates: []tls.Certificate{keyPair},
		Renegotiation: tls.RenegotiateOnceAsClient,
	}, nil
```
Or another constant based on this spec https://pkg.go.dev/crypto/tls#RenegotiationSupport

Reply:
As per the official docs:
> This significantly complicates the state machine and has been the source of numerous, subtle security issues. 

Due to the above, I would rather this is not set by default. Instead, users would need to opt-in (via settings for example).

Reply:
I think this would be better as its own feature request in a separate issue/PR. We'd need to think about how we want this to be opted into and which clients this needs to apply to (eg: should ALL http client make use of that setting, should it be a AuthConfig specific setting, etc).